drivers/gpio/Kconfig.mcp230xx

```diff
@@ -0,0 +1,20 @@
+# MCP23S17 GPIO configuration options
+
+# Copyright (c) 2021 metraTec GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig GPIO_MCP230XX
+	bool "MCP230XX I2C-based GPIO chip"
+	depends on I2C
+	help
+	  Enable driver for MCP230XX I2C-based GPIO chip.
```

Comment:
We can enable the driver "automatically" if relevant compatible is present in DTS. You can check `drivers/gpio/Kconfig.pca953x` for an example. I didn't test the following code but it will look something like
```
# Workaround for not being able to have commas in macro arguments
DT_COMPAT_MICROCHIP_MCP23008 := microchip,mcp23008
DT_COMPAT_MICROCHIP_MCP23017 := microchip,mcp23017
```
and then
```
	default $(dt_compat_enabled,$(DT_COMPAT_MICROCHIP_MCP23008)) || \
		$(dt_compat_enabled,$(DT_COMPAT_MICROCHIP_MCP23017))
```